pkg/discovery/dns/grpc.go

```diff
@@ -68,7 +73,7 @@ func (r *resolver) ResolveNow(_ grpcresolver.ResolveNowOptions) {}
 func (r *resolver) resolve() error {
 	ctx, cancel := context.WithTimeout(r.ctx, r.interval)
 	defer cancel()
-	return r.provider.Resolve(ctx, []string{r.target})
+	return r.provider.Resolve(ctx, []string{r.target}, false)
```

Comment:
Hi @GiedriusS, can you please help me understand this change. This is the only place where we set it to false. Why it is the case?
Can we just enable it to true everywhere?

Reply:
No, we cannot, because in gRPC the same resolver is reused between `--endpoint-group`s. In other words, Build() above is called from multiple places in gRPC but they all reuse the same resolver. They first resolve and then fetch the values from cache. If we flush here then some of the results are lost and the Query component will not connected to some of the endpoints.